### ISSUE TITLE
Add a fixed cost to the calculation

### ISSUE THREAD

**Issue by tzagim** [GuyKh/iec-custom-component#132]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/ludeeus/iec/issues?q=is%3Aissue+label%3A%22Feature+Request%22+).

### Is your feature request related to a problem? Please describe.

סך עלות צריכת חשמל בחשבונית הבאה לא לוקח בחשבון את העלות הקבועה של חלוקה, אספקה ותשלום עבור אמפר

### Describe the solution you'd like

בהגדרה הראשונית אפשרות להגדיר את גודל החיבור, למשל 3X40 שהם 27.71 קילו וואט אמפר (KVA).
בהתבסס על החישוב הזה ניתן לחשב את התשלומים הקבועים.

חישוב עבור KVA
התשלום עבור KVA הוא 3.58 לשנה לפני מע"מ שהם 4.1886 לאחר מע"מ.
ובחישוב שנתי 4.1886X27.17 שווים 116.066
ובחלוקה יומית 0.318

חישוב עבור תשלום חלוקה לתלת פאי הוא 10.25 לפני מע"מ שהם 11.99 כולל מע"מ
תשלום חלוקה לחד פאזי הוא 10.35 כולל מע"מ

תשלום קבוע אספקה להבנתי זה תלוי במספק, אבל במידה וזה תשלום קבוע בהתאם לרשות החשמל מדובר על זה 12.28 לפני מע"מ עבור תלת פאזי שהם 14.367 כולל מע"מ.

[מצ"ב לוח התעריפים של חברת החשמל](https://www.iec.co.il/content/tariffs/contentpages/homeelectricitytariff)

[חישוב קו"א לפי גודל חיבור (עמוד 5)](https://www.gov.il/BlobFolder/policy/12905/he/Files_netunei_hasmal_05129_nisp1.pdf)

### Describe alternatives you've considered

לחלופין ניתן לאפשר להזין את סכום העלות הכוללת החודשית של אספקה וחלוקה, זה לא יתן מענה לעלות הזניחה יחסית של תשלום עבור אמפר.

בנתיים יצרתי לעצמי את התבנית הבאה
```
{% set price = states('sensor.iec_contract_meter_next_bill_electric_forecasted_cost') | float(0) %}
{% set month_days  = states('sensor.days_in_month') | float(0) %}
{% set day = now().day %}
{% set kva_day = 0.318 %}
{% set distribution_pay = 11.99 / month_days %}
{% set delivery_pay = 14.367 / month_days %}
{% if price not in ['Unavailable', 'Unknown', None] %}
    {{ price + (day * (kva_day + distribution_pay + delivery_pay)) }}
{% else %}
  Unavailable
{% endif %}
```
### Additional context
sensor.days_in_month:
```
{% if now().month in [1,3,5,7,8,10,12] %}
  31
{% elif now().month in [4,6,9,11] %}
  30
{% elif now().month == 2 and now().year % 4 == 0 and (now().year % 100 != 0 or now().year % 400 == 0) %}
  29
{% else %}
  28
{% endif %}
```

אשמח לשמוע את חוות דעתך.

כתבתי בעברית מטעמי נוחות.

**Comment by GuyKh** [comment]
@tzagim - this calculation is new to me.
Can you please elaborate on why 3x40 is 27.71 KVA?

**Comment by GuyKh** [comment]
Plus - in your example - `day` is set to be the "day of month" - but many bills are not starting at the 1st of the month.

**Comment by GuyKh** [comment]
As for whether you're at 1 or 3 phase - you can use `get_device_by_device_id` - [see here](https://github.com/GuyKh/py-iec-api/blob/main/iec_api/models/device.py#L34)

**Comment by tzagim** [comment]
@GuyKh 
I started editing and writing that everything can be found in the API.
you mean surely [this one](https://github.com/GuyKh/py-iec-api/blob/main/iec_api/models/device.py#L36)

Information about the calculation and the amounts at [IEC](https://iecapi.iec.co.il/api/content/he-IL/calculators/period), The amounts are slightly different from what appears [in IEC tariffs](https://www.iec.co.il/content/tariffs/contentpages/homeelectricitytariff), I still haven't been able to figure out where the gap comes from.

I'm not an electrician and I don't know the exact calculation.

**Comment by GuyKh** [comment]
You can go with either or

**Comment by tzagim** [comment]
> Plus - in your example - `day` is set to be the "day of month" - but many bills are not starting at the 1st of the month.

I would like to know according to monthly consumption, in addition I understood that if you have a smart meter, suppliers give you a monthly invoice from the first of the month until the end of the month.

**Comment by tzagim** [comment]
@GuyKh 
After an inspection [this is a mistake](https://iecapi.iec.co.il/api/content/he-IL/calculators/period) by IEC and they promised to take care of it.
Is it possible to pull the price data [from here](https://www.iec.co.il/content/tariffs/contentpages/homeelectricitytariff)?

**Comment by GuyKh** [comment]
@tzagim yes. But will require some work. Stay tunes

**Comment by GuyKh** [comment]
Work in progress:
https://github.com/GuyKh/py-iec-api/pull/121

**Comment by GuyKh** [comment]
@tzagim - checkout the PR.
I still think that the calculation is missing the days of bill, and we need to sort out how many days are there since the last invoice

**Comment by tzagim** [comment]
@GuyKh I checked the PR.

For an IEC customer, this is not a problem because you have all the information about the date of the last invoice, billing frequency and the time difference.

For customers of other suppliers, I don't know if they allow flexible dates for the calculation, as far as I understand they all calculate from the first of the month to the first of the following month, if this is not the case, it is possible in the settings to set the day of the calculation each month.

PS, for me, the consumption in the following invoice appears as in the attached file, which does not make sense, and it changes the amount to be paid with the same senselessness. I did not find in the code why this happens. In the meantime, I switched to using monthly consumption.
![attached file](https://github.com/user-attachments/assets/d573cb9f-9169-49de-9229-b2c9b6d6481f)

**Comment by GuyKh** [comment]
The thing is that the `futureConsumption` field is calculated based on the data you requested for (which could be different from month to month etc).

**Comment by tzagim** [comment]
And if in things that don't work properly, yesterday's consumption also appears incorrectly for me.

I tried to reproduce the bug through the IEC API.

Here are my steps:
https://iecapi.iec.co.il/api/Consumption/RemoteReadingRange/00000000000

Chacking for ```totalConsumptionForPeriod```
```py
fromDate = datetime.now() - timedelta(days=1)
```
The Consumption is: "20.888", and Add-on show 2.665.

Chacking for ```futureConsumption``` from ```currentDate``` should show the consumption in the next invoice
As I wrote above, this data also does not appear correctly for me.

Here is an example of how the output looks:

Ask for
```json
{
    "meterSerialNumber": "123465",
    "meterCode": "000",
    "lastInvoiceDate": "2024-07-01",
    "fromDate": "2024-08-01",
    "resolution": 3
}
```

And the response is
```json
{
    "status": 0,
    "futureConsumptionInfo": {
        "lastInvoiceDate": null,
        "currentDate": "2024-08-06",
        "futureConsumption": 1000.999,
        "totalImport": 999.999,
        "totalImportDate": "2024-08-06"
    },
    "fromDate": "2024-08-05",
    "toDate": "2024-08-05",
    "totalConsumptionForPeriod": 20.888,
    "totalImportDateForPeriod": "2024-08-05",
    "meterStartDate": "2024-01-01",
    "totalImport": 988.999,
    "data": [
        {
            "status": 0,
            "date": "2024-08-05T00:00:00.000000",
            "value": 1.082
        },
        {
            "status": 0,
            "date": "2024-08-05T00:15:00.000000",
            "value": 0.585
        },
        {
            "status": 0,
            "date": "2024-08-05T00:30:00.000000",
            "value": 0.615
        }
    ]
}
```

Ask for
```json
{
    "meterSerialNumber": "123465",
    "meterCode": "000",
    "lastInvoiceDate": "2024-07-01",
    "fromDate": "2024-08-05",
    "resolution": 1
}
```

And the response is
```json
{
    "status": 0,
    "futureConsumptionInfo": {
        "lastInvoiceDate": null,
        "currentDate": "2024-08-06",
        "futureConsumption": 1000.999,
        "totalImport": 999.999,
        "totalImportDate": "2024-08-06"
    },
    "fromDate": "2024-08-01",
    "toDate": "2024-08-06",
    "totalConsumptionForPeriod": 300.333,
    "totalImportDateForPeriod": "2024-08-06",
    "meterStartDate": "2024-01-01",
    "totalImport": 999.999,
    "data": [
        {
            "status": 1024,
            "date": "2024-08-01T00:00:00.000000",
            "value": 30.555
        },
        {
            "status": 1024,
            "date": "2024-08-02T00:00:00.000000",
            "value": 40.666
        },
        {
            "status": 1024,
            "date": "2024-08-03T00:00:00.000000",
            "value": 50.777
        },
        {
            "status": 1024,
            "date": "2024-08-04T00:00:00.000000",
            "value": 60.888
        },
        {
            "status": 1024,
            "date": "2024-08-05T00:00:00.000000",
            "value": 90.000
        },
        {
            "status": 0,
            "date": "2024-08-06T00:00:00.000000",
            "value": 1.234
        }
    ]
}

```
PS. Note that the daily consumption data appears in the last line, as of the time the data was retrieved.
I have 0 daily consumption in Add-on and IEC website.

Sorry for the length...

* The data has been replaced with demo numbers

**Comment by GuyKh** [comment]
@tzagim - can you summarize the bottom line since I kinda got lost.

Is the API providing bad data? or is the calculation wrong?

**Comment by tzagim** [comment]
There seems to be a problem with the request and therefore incorrect data is being received from the API.

TL;DR:
1. For yesterday's data "resolution" must to be "1".
2. Daily consumption can be extracted from the same file if it cannot be extracted from another place.
3. There may be an impact to the date as of the invoice date.

**Comment by GuyKh** [comment]
@tzagim I disagree.
Resolution 1 = Daily (hourly), 2 = weekly (day aggregation), and 3 = Monthly (day aggregation)
But the calculation AFAIU is also based on the `fromDate` field (i.e. fromDate = "yesterday" + resolution 1 - would give you yesterday's daily calculation) etc.

If today is Aug 1st, then yesterday would be last month - so I can't do it in a single call no matter how hard I try.

**Comment by tzagim** [comment]
@GuyKh 
First, note that I corrected the mistake, indeed 3 and not 1.

I think the invoice date is significant so [this](https://github.com/GuyKh/iec-custom-component/blob/main/custom_components/iec/coordinator.py#L354) is wrong, because it's entering the same date in ```lastInvoiceDate``` in the request.

It should take the last invoice date from "billingCollection/invoices".

Only If it is the first day of the month, he checks the previous month and so he has yesterday, with the correct ```lastInvoiceDate```

```json
{
    "meterSerialNumber": "123465",
    "meterCode": "000",
    "lastInvoiceDate": "2024-07-10",
    "fromDate": "2024-07-01",
    "resolution": 3
}
```
If that's not the problem, I can't figure out why the Add-on is getting wrong data from the API

**Comment by GuyKh** [comment]
@tzagim - my experience is different.

Here's an example - 
`/billingCollction/invoices` response:
```
                "consumption": xx.0,
                "readingCode": 2,
                "meterReadings": [],
                "fullDate": "2024-07-01T00:00:00",
                "fromDate": "2024-05-13T00:00:00",
                "toDate": "2024-07-10T00:00:00",
                "amountOrigin": xx.49,
                "amountToPay": 0,
                "amountPaid": xx.49,
                "invoiceId": xx,
                "contractNumber": xx,
                "orderNumber": 0,
                "lastDate": "31/07/2024",
                "invoicePaymentStatus": 1,
                "documentID": "1",
                "daysPeriod": 59,
                "hasDirectDebit": false,
                "invoiceType": 0
```

then, when querying for current month (resolution = `3`):
This is the "correct" data. Last invoice date is 
```
{   
    "meterSerialNumber": "{{device_id}}",
    "meterCode": "{{device_code}}",
    "lastInvoiceDate": "2024-07-10",
    "fromDate": "2024-08-03",
    "resolution": 3
}
```

Response is :
```
    "futureConsumptionInfo": {
        "lastInvoiceDate": null,
        "currentDate": "2024-08-07",
        "futureConsumption": 614.211,
        "totalImport": 912.418,
        "totalImportDate": "2024-08-07"
    },
```

When doing the same call with 
```
...
    "lastInvoiceDate": "2024-08-03",
...
```

The response is:
```

Response is :
```
    "futureConsumptionInfo": {
        "lastInvoiceDate": null,
        "currentDate": "2024-08-07",
        "futureConsumption": 614.211,
        "totalImport": 912.418,
        "totalImportDate": "2024-08-07"
    },
```

That's identical

**Comment by GuyKh** [comment]
As for the wrong data from the API - I think the reason could be one of two things:
1. The API itself returns wrong data (this happens quite often)
2. with Future Consumption, I do a lot of fallbacks there one of them could be wrong, but I need a clear example of this

**Comment by tzagim** [comment]
After a long time of debugging, I figured it out.
When the current date is placed in ```fromDate```, it returns incorrect data regarding previous consumption. The date must be yesterday at the latest.

The add-on ask for today...

<details>

<summary>Examples</summary>

Ask for
```json
{
    "meterSerialNumber": "123465",
    "meterCode": "000",
    "lastInvoiceDate": "2024-08-07",
    "fromDate": "2024-08-07",
    "resolution": 3
}
```
Response is :
```json
{
    "status": 0,
    "futureConsumptionInfo": {
        "lastInvoiceDate": null,
        "currentDate": "2024-08-06",
        "futureConsumption": null,
        "totalImport": 111.377,
        "totalImportDate": "2024-08-06"
    },
    "fromDate": "2024-08-01",
    "toDate": "2024-08-07",
    "totalConsumptionForPeriod": 111.695,
    "totalImportDateForPeriod": "2024-08-06",
    "meterStartDate": "xxxx-xx-xx",
    "totalImport": 111.377,
    "data": [
        {
            "status": 1024,
            "date": "2024-08-01T00:00:00.000000",
            "value": 11.983
        },
        {
            "status": 1024,
            "date": "2024-08-02T00:00:00.000000",
            "value": 11.148
        },
        {
            "status": 1024,
            "date": "2024-08-03T00:00:00.000000",
            "value": 11.36
        },
        {
            "status": 1024,
            "date": "2024-08-04T00:00:00.000000",
            "value": 11.316
        },
        {
            "status": 1024,
            "date": "2024-08-05T00:00:00.000000",
            "value": 11.483
        },
        {
            "status": 0,
            "date": "2024-08-06T00:00:00.000000",
            "value": 1.405
        }
    ]
}
```

Ask for
```json
{
    "meterSerialNumber": "123465",
    "meterCode": "000",
    "lastInvoiceDate": "2024-08-06",
    "fromDate": "2024-08-06",
    "resolution": 3
}
```
Response is :
```json
{
    "status": 0,
    "futureConsumptionInfo": {
        "lastInvoiceDate": null,
        "currentDate": "2024-08-07",
        "futureConsumption": 11.91,
        "totalImport": 111.752,
        "totalImportDate": "2024-08-07"
    },
    "fromDate": "2024-08-01",
    "toDate": "2024-08-07",
    "totalConsumptionForPeriod": 100.2,
    "totalImportDateForPeriod": "2024-08-07",
    "meterStartDate": "xxxx-xx-xx",
    "totalImport": 111.752,
    "data": [
        {
            "status": 1024,
            "date": "2024-08-01T00:00:00.000000",
            "value": 11.983
        },
        {
            "status": 1024,
            "date": "2024-08-02T00:00:00.000000",
            "value": 11.148
        },
        {
            "status": 1024,
            "date": "2024-08-03T00:00:00.000000",
            "value": 11.36
        },
        {
            "status": 1024,
            "date": "2024-08-04T00:00:00.000000",
            "value": 11.316
        },
        {
            "status": 1024,
            "date": "2024-08-05T00:00:00.000000",
            "value": 11.483
        },
        {
            "status": 1024,
            "date": "2024-08-06T00:00:00.000000",
            "value": 11.375
        },
        {
            "status": 0,
            "date": "2024-08-07T00:00:00.000000",
            "value": 1.535
        }
    ]
}
```
</details>

**Comment by GuyKh** [comment]
Basically - what your examples show that using `resoultion=3` (monthly) - setting today's date would give wrong number compared with yesterday's date (when the two dates are on the same month). Correct?

**Comment by tzagim** [comment]
> Basically - what your examples show that using `resoultion=3` (monthly) - setting today's date would give wrong number compared with yesterday's date (when the two dates are on the same month). Correct?

Correct

**Comment by GuyKh** [comment]
Have you checked correctness? i.e. is bigger = more correct?

**Comment by tzagim** [comment]
You can't enter a date higher than the current date, yesterday's date gives the information including today.

**Comment by tzagim** [comment]
@GuyKh 
I debugging in the new version.

The following sensors work fine for me:
1. this_month_electric_consumption
2. last_bill_date
3. iec_latest_meter_reading
4. last_bill_electric_usage_to_date
5. last_bill_meter_reading
6. last_bill_length_in_days
7. last_bill_electric_cost

The following sensors are working wrong for me:
1. next_electric_bill_forecasted_cost - Calculated estimated bill: No. of days: 17....  the calculation seems wrong, my last invoice was not on the 25th.
2. yesterday_electric_consumption - it seems to me that because he ask for ```+ delta``` instead of ```- delta```. ```{"meterSerialNumber":"1234567","meterCode":"xxx","lastInvoiceDate":"2024-08-12","fromDate":"2024-08-12","resolution":3}``` returns incorrect information.
3. today_electric_consumption - because of the same problem, for me it only works if you ask for yesterday's date (maybe it's because I have a new counter, I don't know).
4.
